### PR TITLE
[Backport 3.6] Fix submodule error messages

### PR DESCRIPTION
**Pull request by Harry-Ramsey**
## PR checklist

Please remove the segment/s on either side of the | symbol as appropriate, and add any relevant link/s to the end of the line.
If the provided content is part of the present PR remove the # symbol.

- [x] **changelog** not required.
- [x] **development PR** provided: #9833 
- [x] **framework PR** not required.
- [x] **3.6 PR** provided.
- [x] **2.28 PR** not required because: no framework.
- **tests**  provided | not required because: 

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -314,11 +314,11 @@ if(LIB_INSTALL_DIR)
     set(CMAKE_INSTALL_LIBDIR "${LIB_INSTALL_DIR}")
 endif()
 
-if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt")
-    if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git/")
-        message(FATAL_ERROR "${CMAKE_CURRENT_SOURCE_DIR}CMakeLists.txt not found (and does appear to be a git checkout). Run `git submodule update --init` from the source tree to fetch the submodule contents.")
+if (NOT EXISTS "${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt")
+    if (EXISTS "${MBEDTLS_DIR}/.git")
+        message(FATAL_ERROR "${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt not found (and does appear to be a git checkout). Run `git submodule update --init` from the source tree to fetch the submodule contents.")
```
[the remark this answers is not in the document]
Comment (Harry-Ramsey) [reply CMakeLists.txt:319]: If `CMAKE_CURRENT_SOURCE_DIR` is guaranteed in this scenario to be `Mbed TLS` I don't see an issue with dropping that commit. Is this the case?

[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply CMakeLists.txt]: I'm about the last person to ask for CMake knowledge, but [the documentation](https://cmake.org/cmake/help/v3.0/variable/CMAKE_CURRENT_SOURCE_DIR.html) suggests that it's always set, even in very old versions that we don't support anymore.

And when `mbedtls` is embedded in another project that uses CMake, I _think_ this is the right variable to use, as opposed to [`CMAKE_SOURCE_DIR`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SOURCE_DIR.html) which could be a parent project.

[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply CMakeLists.txt]: Can you please just change `.git/` to `.git`? That makes it work on my machine.

[the remark this answers is not in the document]
Comment (eleuzi01) [reply CMakeLists.txt]: Apologies if I'm not understanding something, but I believe `MBEDTLS_DIR` is defined in 3.6, only `MBEDTLS_FRAMEWORK_DIR` isn't. Why not leave `MBEDTLS_DIR` in line 318 in the latest commit? Just curious

**Review by eleuzi01 — APPROVED** [review]
LGTM

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**